apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-credit/footerCreditChoices.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable wpcalypso/import-docblock */
+/* global fullSiteEditing */
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/components';
+
+const CreditTextRenderer = ( { label } ) => <span>{ label }</span>;
+const CreditIconRenderer = props => <Icon { ...props } />;
+
+export const creditOptions = fullSiteEditing.footerCreditOptions;
+
+export const RenderedCreditChoice = ( { choice } ) => {
+	const selection = creditOptions.find( ( { value } ) => value === choice );
+	if ( selection ) {
+		const { renderType, renderProps, label } = selection;
+		const renderer = renderType === 'icon' ? CreditIconRenderer : CreditTextRenderer;
+		return renderer( { label, ...renderProps } );
+	}
+	return null;
```

Comment:
This is very nitpicky for such a small component, but still this would be easier to read with an early return:
```jsx
if ( ! selection ) {
  return null;
}
const { renderType, renderProps, label } = selection;
const renderer = renderType === 'icon' ? CreditIconRenderer : CreditTextRenderer;
return renderer( { label, ...renderProps } );
```

Also what about making it even more explicit?
```jsx
if ( ! selection ) {
  return null;
}
if ( 'icon' === renderType ) {
  return <Icon { ...renderProps } />;
}
return <span>{ label }</span>;
```

Reply:
Sure, I think making this more concise will be better :)